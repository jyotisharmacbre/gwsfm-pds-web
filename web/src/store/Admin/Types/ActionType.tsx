export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
