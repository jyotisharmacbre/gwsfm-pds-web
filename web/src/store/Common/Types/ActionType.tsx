export enum ActionType {
  DEFAULT = 'DEFAULT'
}
