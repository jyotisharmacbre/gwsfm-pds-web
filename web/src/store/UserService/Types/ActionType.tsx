export enum ActionType {
  USER_SERVICE_GET_SUCCESS = 'USER_SERVICE_GET_SUCCESS',
  USER_SERVICE_GET_ERROR = 'USER_SERVICE_GET_ERROR'
}
