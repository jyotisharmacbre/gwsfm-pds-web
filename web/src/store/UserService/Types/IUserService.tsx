export interface IUserServiceData {
  id: string;
  lastName: string;
  firstname: string;
  email: string;
  displayName: string;
  groups: [] | null;
}
