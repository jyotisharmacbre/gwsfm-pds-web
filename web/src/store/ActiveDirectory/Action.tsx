import * as axios from '../../client';
import { ActionType } from './Types/ActionType';
import { Dispatch } from 'redux';

const getADhopSuccess = (response: any) => {
  return {
    type: ActionType.AD_HOP_GET_SUCCESS,
    payload: response
  };
};

const getADhopError = (error: any) => {
  return {
    type: ActionType.AD_HOP_GET_ERROR,
    payload: error
  };
};

const getADpmSuccess = (response: any) => {
  return {
    type: ActionType.AD_PM_GET_SUCCESS,
    payload: response
  };
};

const getADpmError = (error: any) => {
  return {
    type: ActionType.AD_PM_GET_ERROR,
    payload: error
  };
};

const getADpoSuccess = (response: any) => {
  return {
    type: ActionType.AD_PO_GET_SUCCESS,
    payload: response
  };
};

const getADpoError = (error: any) => {
  return {
    type: ActionType.AD_PO_GET_ERROR,
    payload: error
  };
};

export const getADhopOther = (response: any) => {
  console.log(response);
  return {
    type: ActionType.AD_HOP_GET_OTHER,
    payload: response
  };
};

export const getADpmOther = (response: any) => {
  console.log(response);
  return {
    type: ActionType.AD_PM_GET_OTHER,
    payload: response
  };
};

export const getADpoOther = (response: any) => {
  console.log(response);
  return {
    type: ActionType.AD_PO_GET_OTHER,
    payload: response
  };
};

let config = {
  headers: {
    'Content-Type': 'application/json'
  }
};
export const getActiveDirectoryHOP = (searchHOP: string) => {
  return (dispatch: Dispatch) => {
    axios.baseAPI
      .get(
        `/api/DynamicsLookup/getCompanies/${searchHOP}?topCount=50
    `,
        config
      )
      .then(response => {
        dispatch(getADhopSuccess(response.data));
      })
      .catch(error => {
        dispatch(getADhopError(error));
      });
  };
};

export const getActiveDirectoryPM = (searchPM: string) => {
  return (dispatch: Dispatch) => {
    axios.baseAPI
      .get(
        `/api/DynamicsLookup/getCompanies/${searchPM}?topCount=50
    `,
        config
      )
      .then(response => {
        dispatch(getADpmSuccess(response.data));
      })
      .catch(error => {
        dispatch(getADpmError(error));
      });
  };
};

export const getActiveDirectoryPO = (searchPO: string) => {
  return (dispatch: Dispatch) => {
    axios.baseAPI
      .get(
        `/api/DynamicsLookup/getCompanies/${searchPO}?topCount=50
    `,
        config
      )
      .then(response => {
        dispatch(getADpoSuccess(response.data));
      })
      .catch(error => {
        dispatch(getADpoError(error));
      });
  };
};
