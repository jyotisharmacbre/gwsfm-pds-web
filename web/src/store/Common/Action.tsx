import { ActionType } from './Types/ActionType';

export const getDefaultState = () => {
	return {
		type: ActionType.DEFAULT
	};
};
