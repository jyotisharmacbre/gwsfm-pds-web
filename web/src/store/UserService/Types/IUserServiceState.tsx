import { IUserServiceData } from './IUserService';

export interface IUserServiceState {
  currentUserProfile: IUserServiceData;
  userServiceData: Array<IUserServiceData>;
  activityFeedUserServiceData: Array<IUserServiceData>;
  error: string | null;
}
