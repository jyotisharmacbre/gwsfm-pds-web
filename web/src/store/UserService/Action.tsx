import * as axios from '../../client';
import { ActionType } from './Types/ActionType';
import { Dispatch } from 'redux';
import { getUsersForEmailsService } from '../../services';
import { store } from '../index';
import { getDefaultState } from '../Common/Action';

export const getUserNamesForEmailsServiceSuccess = (response: any) => {
	return {
		type: ActionType.USER_NAMES_FOR_EMAILSSERVICE_GET_SUCCESS,
		payload: response
	};
};

export const getUserNamesForEmailsServiceError = (error: any) => {
	return {
		type: ActionType.USER_NAMES_FOR_EMAILS_SERVICE_GET_ERROR,
		payload: error
	};
};

export const getUserNamesForEmailsService = (data: Array<string>) => {
	return (dispatch: Dispatch) => {
		let userData = store.getState().userService.userServiceData;
		let dataNotExists: Array<string> = [];
		data.map((email) => {
			let filter = userData.find((ele) => ele.email.toLowerCase() == email.toLowerCase());
			if (filter == undefined) dataNotExists.push(email);
		});
		if (dataNotExists.length == 0) dispatch(getDefaultState());
		else {
			getUsersForEmailsService(dataNotExists)
				.then((response) => {
					/* istanbul ignore next */ 
					dispatch(getUserNamesForEmailsServiceSuccess(response.data));
				})
				.catch((error) => {
					/* istanbul ignore next */ 
					dispatch(getUserNamesForEmailsServiceError(error));
				});
		}
	};
};
