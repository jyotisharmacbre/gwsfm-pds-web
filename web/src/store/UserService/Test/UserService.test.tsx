
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { getUserNamesForEmailsService, getUserNamesForEmailsServiceSuccess, getUserNamesForEmailsServiceError } from '../Action';
import { ActionType } from './../Types/ActionType';
import { store } from './../../index';

import { IUserServiceState } from '../Types/IUserServiceState';
import nock from 'nock';
import { baseURL } from '../../../client/client';
import userServiceReducer, { initialState } from '../Reducer';

nock(baseURL)
  .get('/api/identity/users/getusernamesforemailids')
  .reply(200, [{ email: 'test1@pds.com' }]);


const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
let _store;
_store = mockStore({
  userServiceData: {
    id: 'string',
    lastName: 'string',
    firstname: 'string',
    email: 'test@pds.com',
    displayName: 'string',
    groups: []
  }
});
const _userService: IUserServiceState = {
  userServiceData: [],
  error: null
};



describe('User service Action', () => {
  it('should not add email if data already exist', () => {
    const getState = jest.mock(
      './../../index'
    );
    let state: any = {
      userService: { userServiceData: [{ email: 'test@pds.com' }] }
    }

    jest
      .spyOn(store, 'getState')
      .mockImplementationOnce(() => state);

    const expectedActions =
    {
      type: ActionType.USER_NAMES_FOR_EMAILSSERVICE_GET_SUCCESS,
      payload: "test data"
    }

    expect(getUserNamesForEmailsServiceSuccess("test data")).toMatchObject(expectedActions);
    expect(getUserNamesForEmailsServiceError("error")).toMatchSnapshot();
    expect(_store.dispatch(getUserNamesForEmailsService(['test@pds.com']))).toMatchSnapshot();
  });

  it('should add email if data not exist', () => {
    const getState = jest.mock(
      './../../index'
    );
    let state: any = {
      userService: { userServiceData: [{ email: 'test@pds.com' }] }
    }

    jest
      .spyOn(store, 'getState')
      .mockImplementationOnce(() => state);
    expect(_store.dispatch(getUserNamesForEmailsService(['test1@pds.com']))).toMatchSnapshot();
  });
});

describe('User service Reducer', () => {
  it('should return updated state on getUSerServiceSuccessAction', () => {
    const getUSerServiceSuccessAction = {
      type: ActionType.USER_SERVICE_GET_SUCCESS,
      payload: {
        userServiceData: [{ email: "test@pds.com" }],
        error: null
      }
    }
    expect(userServiceReducer(initialState, getUSerServiceSuccessAction)).toMatchSnapshot();
  });

  it('should return updated state on getUSerServiceErrorAction', () => {
    const getUSerServiceErrorAction = {
      type: ActionType.USER_SERVICE_GET_ERROR,
      payload: {        
        error: {error: true}
      }
    }
    expect(userServiceReducer(initialState, getUSerServiceErrorAction)).toMatchSnapshot();
  });

  it('should return updated state on getUSerServiceEmailSuccessAction', () => {
    const getUSerServiceSuccessAction = {
      type: ActionType.USER_NAMES_FOR_EMAILSSERVICE_GET_SUCCESS,
      payload: {        
        userServiceData: [{ email: "test@pds.com" }],
        error: null
      }
    }
    expect(userServiceReducer(initialState, getUSerServiceSuccessAction)).toMatchSnapshot();
  });

  it('should return updated state on getUSerServiceEmailErrorAction', () => {
    const getUSerServiceErrorAction = {
      type: ActionType.USER_NAMES_FOR_EMAILS_SERVICE_GET_ERROR,
      payload: {        
        error: {error: true}
      }
    }
    expect(userServiceReducer(initialState, getUSerServiceErrorAction)).toMatchSnapshot();
  });
});