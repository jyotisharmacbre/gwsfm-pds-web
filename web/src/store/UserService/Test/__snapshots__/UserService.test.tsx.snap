// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User service Action should add email if data not exist 1`] = `undefined`;

exports[`User service Action should hide menu onblur 1`] = `
Object {
  "payload": "error",
  "type": "USER_NAMES_FOR_EMAILS_SERVICE_GET_ERROR",
}
`;

exports[`User service Action should hide menu onblur 2`] = `undefined`;

exports[`User service Action should not add email if data already exist 1`] = `
Object {
  "payload": "error",
  "type": "USER_NAMES_FOR_EMAILS_SERVICE_GET_ERROR",
}
`;

exports[`User service Action should not add email if data already exist 2`] = `undefined`;

exports[`User service Reducer should return updated state on getUSerServiceEmailErrorAction 1`] = `
Object {
  "error": Object {
    "error": Object {
      "error": true,
    },
  },
  "userServiceData": Array [],
}
`;

exports[`User service Reducer should return updated state on getUSerServiceEmailSuccessAction 1`] = `
Object {
  "error": null,
  "userServiceData": Object {
    "error": null,
    "userServiceData": Array [
      Object {
        "email": "test@pds.com",
      },
    ],
  },
}
`;

exports[`User service Reducer should return updated state on getUSerServiceErrorAction 1`] = `
Object {
  "error": Object {
    "error": Object {
      "error": true,
    },
  },
  "userServiceData": Array [],
}
`;

exports[`User service Reducer should return updated state on getUSerServiceSuccessAction 1`] = `
Object {
  "error": null,
  "userServiceData": Object {
    "error": null,
    "userServiceData": Array [
      Object {
        "email": "test@pds.com",
      },
    ],
  },
}
`;
